CDTDatastoreTests/CDTSessionCookieInterceptorTests.m

```diff
@@ -324,4 +356,146 @@ - (void)testCookieInterceptorRenewsEarly
+            NSString *httpBody = [[NSString alloc] initWithData: request.OHHTTPStubs_HTTPBody encoding:NSUTF8StringEncoding];
+            XCTAssert([httpBody isEqualToString:@"name=username403&password=password"]);
+            return [OHHTTPStubsResponse
+                    responseWithJSONObject:@{
+                                             @"ok" : @(YES),
+                                             @"name" : @"username",
+                                             @"roles" : @[ @"_admin" ]
+                                             }
+                    statusCode:200
+                    headers:@{
+                              @"Set-Cookie" : [NSString
+                                               stringWithFormat:@"%@; Version=1; Path=/; HttpOnly; Max-Age=86400",
+                                               testCookieHeaderValue2]
+                              }];
+        }];
+        
+        // get resource successfully using new cookie
+        [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+            XCTAssert([testCookieHeaderValue2 isEqualToString: request.allHTTPHeaderFields[@"Cookie"]]);
+            XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+            return [OHHTTPStubsResponse responseWithJSONObject:@{@"ok" : @(YES)} statusCode:200 headers:@{}];
+        }];
+    } else {
+        // get resource fails, cookie was not re-newed
+        [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+            XCTAssert([testCookieHeaderValue isEqualToString: request.allHTTPHeaderFields[@"Cookie"]]);
+            XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+            return [OHHTTPStubsResponse responseWithJSONObject:@{@"error": error,
+                                                                 @"reason": reason}
+                                                    statusCode:403 headers:@{}];
+        }];
+    }
+
+    
+    [helper doStubsForHost:@"username403.cloudant.com"];
+    
+    CDTSessionCookieInterceptor *interceptor =
+    [[CDTSessionCookieInterceptor alloc] initWithUsername:@"username403" password:@"password"];
+    // create a context with a request which we can use
+    NSURL *url = [NSURL URLWithString:@"http://username403.cloudant.com/somedb"];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url];
+    
+    CDTURLSession *session = [[CDTURLSession alloc] initWithCallbackThread:[NSThread currentThread] requestInterceptors:@[interceptor] sessionConfigDelegate: nil];
+    
+    // Initial GET will get a cookie and resource
+    CDTURLSessionTask *task = [session dataTaskWithRequest:request taskDelegate:nil];
+    [task resume];
+    while ([task state] != NSURLSessionTaskStateCompleted) {
+        [NSThread sleepForTimeInterval:1.0];
+    }
+    
+    // Second GET should fail with 403 error
```

Comment:
This is not true, the second `GET` (task2) has different results depending on the path.
expired creds -> (under the covers) 403, renew cookie -> 200
non-expired -> 403

It would probably be valuable to assert on the response code for each path.